src/NextGen/batchManager.ts

```diff
@@ -0,0 +1,52 @@
+import {
+    Batch,
+    BatchEnvironmentEnum,
+    BaseEvent,
+} from '@mparticle/event-models';
+import { BaseMPMessage } from './core/baseMPMessage';
+import { isEmpty } from '../utils';
+
+// TODO: Sync this with existing API Client(s)
+export interface IMpApiClientImplementation {
+    uploadBatch(batch: Batch): void;
+}
+
+export default class BatchManager {
+    constructor(private api: IMpApiClientImplementation) {
+        this.api = api;
+    }
+
+    // QUESTION: Should this take an event or message as a param?
+    // Making an assumption that creating a batch should require a message
+    public createBatch(events: BaseMPMessage[]): Batch {
+        // QUESTION: In the existing Batch Uploader, we have a method to create new batches based on events
+        //           should this method reuse that code or will this be a much simpler create method?
+
+        if (!isEmpty(events)) {
+            return {
+                environment: BatchEnvironmentEnum.unknown,
+                mpid: 'test-mpid',
+                events: (events as unknown) as BaseEvent[],
+            };
+        } else {
+            return null;
+        }
+    }
+
+    // QUESTION: We should consider making this take an array or queue of batches, as we are required to upload batches in sequence if they are queued
+    public uploadBatch(batch: Batch): void {
```

Comment:
I guess the question boils down to whether we should have implementation code that loops through the array/queue and calls this method or whether we should have this method just take an array/queue and have the logic there?

Reply:
I think the answer depends on where we want to hold the "business logic" for uploading batches, since sequencing is a core piece of functionality. I think there's a disconnect on my part regarding the difference between the "BatchManager" and "UploadStrategy" we discussed.

Reply:
I think this code should just upload the batch and the queuing should occur on the level that's calling this method.